src/hooks/users/useWorldUsers.tsx

```diff
@@ -41,44 +41,38 @@ export const WorldUsersProvider: React.FC<WorldUsersProviderProps> = ({
     venueId,
   });
 
-  const shouldConnect =
-    !isSovereignVenueLoading &&
-    isDefined(sovereignVenueId) &&
-    isDefined(venueId);
-
-  // TODO: refactor this to not use useFirestoreConnect
-  useFirestoreConnect(() => {
-    if (!shouldConnect) return [];
-
-    const relatedLocationIds = [venueId];
-
-    if (sovereignVenueId) {
-      relatedLocationIds.push(sovereignVenueId);
+  const relatedLocationIds: string[] | undefined = useMemo(() => {
+    if (isSovereignVenueLoading || !venueId || !sovereignVenueId) return;
+
+    return [sovereignVenueId, venueId];
+  }, [isSovereignVenueLoading, sovereignVenueId, venueId]);
+
+  // TODO: should we refactor this to use useWorldUsersQuerySubscription rather than connecting it to receive data updates?
+  //   It seems that useWorldUsersQuerySubscription doesn't return originalArgs, so we can't use it in the same way
+  //   as we are currently doing things.. but we could potentially just pass the same args object into the hook as what we pass
+  //   down for the descendant code to use.. which would work around that 'limitation'
+  // TODO: https://redux-toolkit.js.org/rtk-query/usage/queries#selecting-data-from-a-query-result
+  const { originalArgs: worldUsersApiArgs } = useWorldUsersQuery(
+    relatedLocationIds ? { relatedLocationIds } : skipToken,
```

Comment:
can we move this ternary expression to a const and use here?

Reply:
For what purpose? Just moving it out of the params, or?